report/validation/thermodesorption_spectra/dark/dark_tds.md

```diff
@@ -0,0 +1,256 @@
+
+# boundary conditions
+model.boundary_conditions = [F.DirichletBC(surfaces=[1, 2], value=0, field="solute")]
+
+# temperature
+start_tds = t_imp + t_rest  # s
+min_temp, max_temp = 300, 1000
+
+model.T = F.Temperature(
+    value=sp.Piecewise(
+        (T_imp, F.t < t_imp),
+        (T_rest, F.t < start_tds),
+        (min_temp + Beta * (F.t - start_tds), True),
+    )
+)
+
+# trap settings
+k_0 = D_0 / (1.1e-10**2 * 6 * w_atom_density)
+damage_dist = 1 / (1 + sp.exp((F.x - 2.5e-06) / 5e-07))
+n_i = np.array([2.4e-3, 4.8, 3.8, 2.6, 3.6, 1.1]) * 1e25
+neutron_induced_traps = []
+E_p = [1.04, 1.15, 1.35, 1.65, 1.85, 2.05]
+
+trap_1 = F.Trap(
+    E_k = 0.28,
+    k_0 = k_0,
+    E_p = 1.04,
+    p_0 = 1e13,
+    density = n_i[0],
+    materials = damaged_tungsten
+)
+
+for i in range(1, 6):
+    neutron_induced_traps.append(F.Trap(
+        k_0 = k_0,
+        E_k = E_D,
+        p_0 = 1e13,
+        E_p = E_p[i],
+        density = n_i[i] * damage_dist,
+        materials = damaged_tungsten,
+    ))
+
+traps = [trap_1] + neutron_induced_traps
+model.traps = traps
+
+model.dt = F.Stepsize(
+    initial_value=1,
+    stepsize_change_ratio=1.1,
+    t_stop=t_imp + t_rest * 0.5,
+    dt_min=1e-1,
+    stepsize_stop_max=50,
+)
+
+model.settings = F.Settings(
+    absolute_tolerance=1e10,
+    relative_tolerance=1e-10,
+    final_time=start_tds + (max_temp - min_temp) / Beta,  # time to reach max temp
+)
+
+derived_quantities = F.DerivedQuantities(
+    [
+        F.AverageVolume("T", volume=1),
+        F.TotalVolume("solute", volume=1),
+        F.TotalVolume("retention", volume=1),
+        F.TotalVolume("1", volume=1),
+        F.TotalVolume("2", volume=1),
+        F.TotalVolume("3", volume=1),
+        F.TotalVolume("4", volume=1),
+        F.TotalVolume("5", volume=1),
+        F.TotalVolume("6", volume=1),
+        F.HydrogenFlux(surface=1),
+        F.HydrogenFlux(surface=2),
+    ],
+)
+
+model.exports = [derived_quantities]
+
+model.initialise()
+model.run()
+```
+
+```{note}
+Implementation details from {cite}`dark_modelling_2024_code`.
+```
+
++++
+
+## Comparison with experimental data
+
+The results produced by FESTIM are in good agreement with the experimental data. The grey areas represent the contribution of each trap to the global TDS spectrum.
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+t = derived_quantities.t
+flux_left = derived_quantities.filter(fields="solute", surfaces=1).data
+flux_right = derived_quantities.filter(fields="solute", surfaces=2).data
+flux_total = -np.array(flux_left) - np.array(flux_right)
+
+t = np.array(t)
+temp = min_temp + Beta * (t - start_tds)
+
+# plotting simulation data
+plt.plot(temp, flux_total, linewidth=3, label="FESTIM")
+
+# plotting trap contributions
+trap_data = [derived_quantities.filter(fields=f"{i}").data for i in range(1, 7)]
+contributions = [-np.diff(trap) / np.diff(t) for trap in trap_data]
+
+colors = [(0.9*(i % 2), 0.2*(i % 4), 0.4*(i % 3)) for i in range(6)]
```

Comment:
Just sharing an awesome tool made by @holtzy: [PyPalettes](https://python-graph-gallery.com/color-palette-finder/) which helps finding cool colour palettes!

I quite like [this one](https://python-graph-gallery.com/color-palette-finder/?palette=acadia) for instance.

The python code for this is:

```python
from pypalettes import load_cmap
cmap = load_cmap("Acadia")
```

Reply:
That is very cool, thanks!